src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +27,177 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
+            Flags_UnsetSwitch(globalCtx, this->unk_168);
+            this->actionFunc = BgYdanHasi_initWater;
+        }
+        func_8002F948(&this->dyna.actor, 0x205E);
+    } else {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y - 47.0f, 0.5f)) {
+            this->actionFunc = BgYdanHasi_decWaterTimer;
+        }
+        func_8002F948(&this->dyna.actor, 0x205E);
+    }
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    globalCtx->colCtx.stat.colHeader->waterBoxes[1].unk_02 = this->dyna.actor.posRot.pos.y;
+    if (true) {}
+    return waterBox + 0x1;
+}
+
+void BgYdanHasi_decWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
+    if (this->timer == 0) {
+        this->actionFunc = BgYdanHasi_updateWater;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE690.s")
+void BgYdanHasi_setupThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk_168) != 0) {
+        this->timer = 260;
+        this->dyna.actor.draw = BgYdanHasi_Draw;
+        this->actionFunc = BgYdanHasi_updateThreeBlocks;
+        func_800800F8(globalCtx, 0xBE0, 0x1E, &this->dyna.actor, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE7C8.s")
+void BgYdanHasi_updateThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    BgYdanHasi* temp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE810.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 3.0f) != 0) {
+            Flags_UnsetSwitch(globalCtx, this->unk_168);
+            this->dyna.actor.draw = NULL;
+            this->actionFunc = BgYdanHasi_setupThreeBlocks;
+            return;
+        }
+        func_8002F948(&this->dyna.actor, 0x2024);
+        return;
+    }
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y + 120.0f, 3.0f) == 0) {
+        func_8002F948(&this->dyna.actor, 0x2024);
+        return;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE8DC.s")
+void BgYdanHasi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanHasi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE930.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE99C.s")
+void BgYdanHasi_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    char pad[0xE];
+    GraphicsContext* gfxCtx;
+    Gfx* sp50;
```

Comment:
The Gfx* array expected by Graph_OpenDisps and Graph_CloseDisps is normally called dispRefs.
```suggestion
    GraphicsContext* gfxCtx;
    Gfx* dispRefs[4];
```